lib/get-file-path-from-url.js

```diff
@@ -8,21 +8,29 @@ const path = require('path');
  * @param {object} [pathLib=path] - path library, override for testing
  * @return {string} - will return 'dummy' if the path is bad
  */
-function getFilePathFromUrl(url, basePath, { pathLib = path } = {}) {
+function getFilePathFromUrl(url, basePath, { pathLib = path, baseHref = '' } = {}) {
   if (!basePath) {
     throw new Error('basePath must be specified');
   }
   if (!pathLib.isAbsolute(basePath)) {
     throw new Error(`${basePath} is invalid - must be absolute`);
   }
 
-  const relativePath = url.split('?')[0];
+  let relativePath = url.split('?')[0];
 
   if (relativePath.indexOf('../') > -1) {
     // any path attempting to traverse up the directory should be rejected
     return 'dummy';
   }
 
+  if (baseHref) {
```

Comment:
just to check I understand here. where angular has a baseHref, it will add this to all the url requests, and you therefore need to remove them for testing purposes?

Reply:
The app is served as part of a bigger site under `/myapp/`, so index.html contains 
`<base href="myapp">`
To get this working properly with angular-http-server the base part needs to be stripped of from every url when looking for a file to load. Otherwise the browser will serve `index.html` when the browser tries to load for example `localhost:8080/myapp/favicon.ico`.

A use case for this can be easily created using the angular cli:
```
ng new myapp
cd myapp
ng build --prod --base-href myapp
angular-http-server --path dist/myapp
```
Without `--baseHref myapp` the app will not load under `localhost:8080/myapp`. It may seem to load correctly under `localhost:8080` but a reload will fail.